autoload/javacomplete.vim

```diff
@@ -121,7 +113,7 @@ function! javacomplete#GetBase(extra)
 endfunction
 
 function! s:GetBase(extra)
-  let base = expand("~". g:FILE_SEP. ".javacomplete2". g:FILE_SEP. a:extra)
+  let base = expand(g:JavaComplete_BaseDir. g:FILE_SEP. a:extra)
```

Comment:
@artur-shaik if you want to use `javacomplete2` instead of `.javacomplete`, I think we should change the default BaseDir to `~/.cache/javacomplete2`